index-import.cpp

```diff
@@ -376,15 +382,23 @@ int main(int argc, char **argv) {
   const size_t length = InputIndexPaths.size();
   const size_t numStrides = ((length - 1) / stride) + 1;
 
+  static std::mutex lock{};
```

Comment:
What about using a dispatch semaphore (value = 1) instead? Seems to be better for consistency although I'm not sure of the performance implications

Reply:
Good point, thanks. It didn't even cross my mind 🤦‍♂️. We could measure perf, but as I mentioned I don't think verbose mode will be used in performance critical cases.

Reply:
Done